File: client/web/sudoo/lib/domain/model/user/gender.dart
import 'package:json_annotation/json_annotation.dart';

enum Gender {
  @JsonValue("MALE")
  male("Male"),
  @JsonValue("FEMALE")
  female("Female"),
  @JsonValue("OTHER")
  other("Other");

  final String value;

  const Gender(this.value);

  factory Gender.fromValue(String value) {
    switch(value) {
      case "Male":
        return male;
      case "Female":
        return female;
      default:
        return other;
    }
  }
}

File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/domain/entity/user/Gender.kt
package com.sudo248.sudoo.domain.entity.user

enum class Gender(val value: String) {
    MALE("Nam"),
    FEMALE("Nữ"),
    OTHER("Khác");

    companion object {
        fun fromStringValue(value: String): Gender {
            return when(value){
                "Nam" -> MALE
                "Nữ" -> FEMALE
                else -> OTHER
            }
        }
    }
}